src/onelogin/saml2/response.py

```diff
@@ -386,7 +383,11 @@ def validate_num_assertions(self):
         """
         encrypted_assertion_nodes = OneLogin_Saml2_Utils.query(self.document, '//saml:EncryptedAssertion')
         assertion_nodes = OneLogin_Saml2_Utils.query(self.document, '//saml:Assertion')
-        return (len(encrypted_assertion_nodes) + len(assertion_nodes)) == 1
+        num_assertions = len(encrypted_assertion_nodes) + len(assertion_nodes)
+        if num_assertions == 1:
+            return True
+        else:
+            raise Exception('Could not validate response: expected exactly one assertion; got {}'.format(num_assertions))
```

Comment:
I don't think this will be particularly useful, because the number of assertions is usually either 0 or 1 as far as I've seen. Doesn't hurt though :)